File: Frontend/src/types/device.ts
// Mirrors the state MediaPipe (camera/hands/face) and the audio pipeline
// (microphone/speaker) will report once integrated.
export type DeviceState = 'ready' | 'tracking' | 'listening' | 'warning' | 'offline'

export interface DeviceStatus {
  camera: DeviceState
  hands: DeviceState
  face: DeviceState
  microphone: DeviceState
  speaker: DeviceState
  ai: DeviceState
  lightLevel: DeviceState
}

File: Backend/app/data/device.py
from app.models.device import DeviceStatus

# Seeded with the same values as src/mocks/deviceStatus.json. Once
# MediaPipe is wired up, GET /api/device/status should return the most
# recent snapshot the ML pipeline produced instead of this static object.
DEVICE_STATUS = DeviceStatus(
    camera="tracking",
    hands="tracking",
    face="tracking",
    microphone="listening",
    speaker="ready",
    ai="ready",
    lightLevel="warning",
)
